src/lib/containers/download_list/DownloadConfirmation.tsx

```diff
@@ -153,13 +153,21 @@ export const DownloadConfirmation: React.FunctionComponent<DownloadConfirmationP
     fileCount: 0,
     fileSize: 0
   })
-  const [queryOrFolder] = useState<QueryOrFolder>({
-    queryBundleRequest: getLastQueryRequest!(),
+  const lastQueryRequest = getLastQueryRequest!()
```

Comment:
is `getLastQueryRequest` memoized? If not, `lastQueryRequest` may change on each render and would kill performance with your effect below

Reply:
Verified new code in useDeepCompareEffect() is called only when I'd expect it (which is very rare).